### ISSUE TITLE
Should I place my api routes under [locale] ?

### ISSUE THREAD

**Issue by dBianchii** [QuiiBz/next-international#389]
I am using this library, and I managed to get this to work when my api folder is not inside the [locale] folder. Is this the correct way to do it?

If I don't have locale accessible inside the api, this will mean I won't have access to the user's selected language in there, yes? I'm not sure if this will be a problem

**Comment by gustaveWPM** [comment]
![image](https://github.com/QuiiBz/next-international/assets/118654172/fab83568-caf7-4d97-821c-886814e0d134)

> If I don't have locale accessible inside the api, this will mean I won't have access to the user's selected language in there, yes? I'm not sure if this will be a problem

I guess you should add the locale in headers/parameters of your API endpoint, then?

**Comment by kejiahp** [comment]
I don't believe it's necessary to place your API routes inside the [locale] folder.
You can access the locale via the `NEXT_LOCALE` cookie or the `Accept-Language` header an if that won't do you can also access the locale by formatting the URL gotten from the `Referer` header.

**Comment by dBianchii** [comment]
@kejiahp @QuiiBz Do you know how I can do it? I want to have my error messages be translated.

**Comment by kejiahp** [comment]
@dBianchii I would assume you are trying to access the `t` object on the server-side, you can do this using the `getTranslations()` function from `next-intl`.

[Read more: getTranslations](https://next-intl-docs.vercel.app/docs/environments/server-client-components#async-components)

If my assumption isn't right, kindly correct me and provide more context on your problem.

**Comment by dBianchii** [comment]
@kejiahp this is in the next-international repo
